# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1684876495 -7200
# Parent  ea6e0e94614e54741f8b54273d49fc5c1d63938a
Misc: Make update check take policy DisableAppUpdate into account.

diff --git a/mail/base/content/aboutDialog.js b/mail/base/content/aboutDialog.js
--- a/mail/base/content/aboutDialog.js
+++ b/mail/base/content/aboutDialog.js
@@ -101,18 +101,27 @@ async function init(aEvent) {
       screen.availWidth / 2 - window.outerWidth / 2,
       screen.availHeight / 5
     );
   }
 }
 
 async function BetterbirdUpdateCheck(win) {
   let needsUpdate = "arguments" in win && win.arguments[0].needsUpdate;
+  let checkForUpdate = !needsUpdate;
   let cantCheck = false;
-  if (!needsUpdate) {
+  if (Services.policies && !Services.policies.isAllowed("appUpdate")) {
+    console.info(
+      "Betterbird skipping update check due to policy DisableAppUpdate"
+    );
+    checkForUpdate = false;
+    cantCheck = true;
+  }
+
+  if (checkForUpdate) {
     try {
       let response = await fetch(
         "https://www.betterbird.eu/start/versions.txt",
         { cache: "no-store" }
       );
       if (!response.ok) {
         console.error("Betterbird update check FAILED");
         cantCheck = true;
diff --git a/mail/components/MailGlue.jsm b/mail/components/MailGlue.jsm
--- a/mail/components/MailGlue.jsm
+++ b/mail/components/MailGlue.jsm
@@ -745,16 +745,23 @@ MailGlue.prototype = {
           tabParams,
         }
       );
       linkHandled.data = true;
     }
   },
 
   async BetterbirdUpdateCheck() {
+    if (Services.policies && !Services.policies.isAllowed("appUpdate")) {
+      console.info(
+        "Betterbird skipping update check due to policy DisableAppUpdate"
+      );
+      return;
+    }
+
     let lastCheckTime = Services.prefs.getIntPref(
       "betterbird.update.lastCheckTime",
       0
     );
     let now = Math.round(Date.now() / 1000);
     // We check once a week. 604800 = 7 * 24 * 60 * 60.
     if (now - lastCheckTime < 604800) {
       console.info(
